### PR TITLE
Revert TokenNetworkUtils library

### LINKED ISSUE #1434: Update etherscan verification code for linked libraries
Pulling this out of https://github.com/raiden-network/raiden-contracts/pull/1428

We need to update the verification code to handle linked libraries.

### Refs
- https://ropsten.etherscan.io/apis#contracts

### LINKED ISSUE #1435: Move further code from TokenNetwork to library
in https://github.com/raiden-network/raiden-contracts/pull/1428 only a couple of functions were pulled out to minimize the diff.

### Tasks
- Go through all remaining functions and see whether they can be moved to the library.
- Try moving `structs`/`enum`s to the library, see if it reduces contract code size.

### PR DESCRIPTION
**Pull request by palango**
Fixes https://github.com/raiden-network/raiden-contracts/issues/1503
Fixes #1434 
Fixes #1435 

Revert the addition of the `TokenNetworkUtils` library
This eases
- deployment and the code in this package
- usage and verification of the contracts in other packages (`raiden` itself)

## Contract size changes

Those seem mostly related to the disabled yul optimizer. But we add it still fits we can add those back later.

Before:
```
running compile_contracts
TokenNetworkUtils 896 3.6458333333333335%
TokenNetwork 17217 70.05615234375%
TokenNetworkRegistry 20515 83.47574869791666%
```

After:
```
running compile_contracts
TokenNetwork 19055 77.53499348958334%
TokenNetworkRegistry 23336 94.95442708333334%
```

### What's tricky about this PR (if any)

----

Any reviewer can check these:

* [ ] If the PR is fixing a bug or adding a feature, add an entry to CHANGELOG.md.
* [ ] If the PR changed a Solidity source, run `make compile_contracts` and add the resulting `raiden_contracts/data/contracts.json` in the PR.
* [ ] If the PR is changing documentation only, add `[skip ci]` in the commit message so Travis does not waste time.
    * [ ] But, if the PR changes comments in a Solidity source, do not add `[skip ci]` and let Travis check the hash of the source.
* [ ] In Python, use keyword arguments
* [ ] Squash unnecessary commits
* [ ] Comment commits
* [ ] Follow naming conventions
    * `solidityFunction`
    * `_solidity_argument`
    * `solidity_variable`
    * `python_variable`
    * `PYTHON_CONSTANT`
* [ ] Follow the Signature Convention in [CONTRIBUTING.md](./CONTRIBUTING.md)
* For each new contract
    * [ ] The deployment script deploys the new contract.
    * [ ] `etherscan_verify.py` runs on the new contract.
* Bookkeep
    * [ ] The gas cost of new functions are stored in `gas.json`.
* Solidity specific conventions
    * [ ] Document arguments of functions in natspec
    * [ ] Care reentrancy problems
    * if the PR adds or removes `require()` or `assert()`
        * [ ] add an entry in Changelog
        * [ ] open an issue in the client, light client, service repos so the change is reflected there
        * Just adding a message in `require()` doesn't require these steps.
* [ ] When you catch a require() failure in Solidity, look for a specific error message like `pytest.raises(TransactionFailed, match="error message"):`

And before "merge" all checkboxes have to be checked.  If you find redundant points, remove them.

### CODE REVIEW

File: raiden_contracts/contract_source_manager.py [hunk 1]
```diff
@@ -174,6 +175,9 @@ def check_runtime_codesize(d: Dict) -> None:
     """Raises a RuntimeError if the runtime codesize exceeds the EIP-170 limit"""
     for name, compilation in d.items():
         runtime_code_len = len(compilation["bin-runtime"]) // 2
+
+        if "TokenN" in name:
```
Comment (palango) [inline raiden_contracts/contract_source_manager.py:179]: Needs to be removed, just to check the size changes

File: raiden_contracts/data/source/raiden/TokenNetwork.sol [hunk 2]
```diff
@@ -1589,6 +1574,165 @@ contract TokenNetwork is Utils, Controllable {
         delete participants_hash_to_channel_identifier[pair_hash];
     }
 
+        function recoverAddressFromBalanceProof(
```
Comment (istankovic) [inline raiden_contracts/data/source/raiden/TokenNetwork.sol:1577]: This seems indented one level more than it needs to be.

File: raiden_contracts/data/source/test/TokenNetworkInternalsTest.sol [hunk 3]
```diff
@@ -261,17 +255,17 @@ contract TokenNetworkUtilsTest is TokenNetwork {
 
     function minPublic(uint256 a, uint256 b) public view returns (uint256)
     {
-        return TokenNetworkUtils.min(a,b);
+        return min(a,b);
```
Comment (istankovic) [inline raiden_contracts/data/source/test/TokenNetworkInternalsTest.sol:258]: I'd put a space after the comma, to be consistent.

File: raiden_contracts/data/source/test/TokenNetworkInternalsTest.sol [hunk 4]
```diff
@@ -261,17 +255,17 @@ contract TokenNetworkUtilsTest is TokenNetwork {
...
     }
 
     function failsafe_subtractPublic(uint256 a, uint256 b) public view returns (uint256, uint256)
     {
-        return TokenNetworkUtils.failsafe_subtract(a,b);
+        return failsafe_subtract(a,b);
```
Comment (istankovic) [inline raiden_contracts/data/source/test/TokenNetworkInternalsTest.sol:263]: Same here.

File: raiden_contracts/deploy/etherscan_verify.py [hunk 5]
```diff
@@ -177,7 +178,7 @@ def etherscan_verify_contract(
         "contractname": contract_name,
         "compilerversion": "v" + metadata["compiler"]["version"],
         # 0 = Optimization used, 1 = No Optimization
-        "optimizationUsed": 0 if metadata["settings"]["optimizer"]["enabled"] is False else 1,
+        "optimizationUsed": 0 if optimization_details is not None else 1,
         "runs": metadata["settings"]["optimizer"]["runs"],
         # Typo is intentional. Etherscan does not like the correct spelling.
         "constructorArguements": constructor_args,
```
Comment (istankovic) [inline raiden_contracts/deploy/etherscan_verify.py:184]: Commit message has a typo, I think:

> but we have to test the on a testchain

(test the on)

**Review by istankovic — APPROVED** [review]
Looks fine to me.

**Comment by istankovic** [comment]
This will also fix raiden-network/raiden#7203, at least in the sense that no chages are needed then on the raiden side. However, we'd still like to use a single mechanism to deploy contracts, i.e. the `ContractDeploy` from `raiden-contracts`.

File: raiden_contracts/deploy/etherscan_verify.py (hunk 5 above)
Comment (palango) [reply raiden_contracts/deploy/etherscan_verify.py:184]: 👀  Fixed!

**Pull request merged**